tests/unit/test_charm.py

```diff
@@ -46,23 +45,36 @@ class TestCharm(unittest.TestCase):
     DETACH_ACTION_PARAMS = {"usim-imsi": None, "usim-opc": None, "usim-k": None}
 
     def setUp(self) -> None:
-        self.remote_app_name = "magma-access-gateway-operator"
-        self.relation_name = "lte-core"
         self.harness = testing.Harness(SrsLteCharm)
         self.addCleanup(self.harness.cleanup)
+        self.maxDiff = None
         self.harness.begin()
-        self.peer_relation_id = self.harness.add_relation("replicas", self.harness.charm.app.name)
-        self.harness.add_relation_unit(self.peer_relation_id, self.harness.charm.unit.name)
-        self.harness.set_leader(True)
+
+    def create_lte_core_relation(self) -> int:
+        relation_name = "lte-core"
+        remote_app_name = "magma-access-gateway-operator"
+        mme_ipv4_address = "1.2.3.4"
+        relation_data = {"mme_ipv4_address": mme_ipv4_address}
+        relation_id = self.harness.add_relation(
+            relation_name=relation_name, remote_app=remote_app_name
+        )
+        self.harness.update_relation_data(
+            relation_id=relation_id,
+            app_or_unit=remote_app_name,
+            key_values=relation_data,
+        )
+        return relation_id
 
     @patch("builtins.open", new_callable=mock_open)
     @patch("os.mkdir")
     @patch("shutil.copy")
     @patch("shutil.rmtree")
     @patch("subprocess.run")
-    def test_given_list_of_packages_to_install_when_install_then_apt_cache_is_updated(
+    def test_given_unit_is_leader_when_install_then_apt_cache_is_updated(
         self, patch_subprocess_run, _, __, ___, ____
```

Comment:
You can get rid of all those underscore if you add `new=Mock` to the patches. IE: `@patch("os.mkdir", new=Mock)`. This is valid for the whole file.

Reply:
Done

Reply:
The parameter to new should be a `Callable`. You should pass the name of the class only and not call it: `new=Mock`. With `Mock()`, it still works because calling an instance of `Mock` returns a `Mock`. This creates unnecessary instances.

```
>>> from unittest.mock import Mock
>>> a = Mock()
>>> b = a()
>>> a is b
False
```

Reply:
Damn I merged too fast, thanks for letting me know, I'll add those in a subsequent PR

Reply:
The change will be [here](https://github.com/charmed-osm/srs-enb-ue-operator/pull/17)